add get_pk_from_address to eth_bridge

gaieges opened the pull request:
This logic should live here instead of hmt-servers or loader.

- [x] Still need doctests before merge
- [x] Needs kvstore deployed to local blockchain in order to finish the above

Note: renames `get_pk_from_address` to `get_pub_key_from_addr` and adds a corresponding SET function: `set_pub_key_at_addr`

Fixes #86

gaieges commented:
Welp .. @rbval and I realized that we will need a method to deploy the kvstore contract into the local blockchain in order to properly test this via the doctest (similar to the way we do with hmt-token) .. so we're blocked on that..

gaieges commented:
Yep!

gaieges commented:
Should be good now, I'm sure a test will break that i'll fix

rbval commented on hmt_escrow/eth_bridge.py:
```diff
@@ -232,6 +238,46 @@ def deploy_factory(gas: int = GAS_LIMIT, **credentials) -> str:
+        raise ValueError('environment variable GAS_PAYER required')
+    ValueError: environment variable GAS_PAYER required
+    >>> os.environ['GAS_PAYER'] = Web3.toChecksumAddress('0x3895d913a9a231d2b215f402c528511b569c676d')
+    >>> get_pk_from_address(Web3.toChecksumAddress('0xc23546d9c10e85322cb63112ec034090b407a019'))
```
This doesn't really test that we fetched the correct public key.

We should probably add a "set" method as well so we can test that function actually gives us the correct public key.

Also would rename it from `get_pk_from_address` to `get_pub_key_from_addr`.

gaieges replied:
Ya I can do those renames .. Also this was from the other day when the tests didn't work, fixing them up now.  Was just trying to figure out how to do a test set of a kv pair, but I don't think testing that or the function should live in this repo .. we should use the api of kvstore

rbval replied:
Well our kvstore returns a string when we do the get. `get_pub_key_from_addr` converts it to bytes so we add a bit functionality there which would be good to test. Can't count the amount of times and hours wasted because I was expecting bytes or str and vice versa.

rbval commented on hmt_escrow/eth_bridge.py:
```diff
@@ -232,6 +238,46 @@ def deploy_factory(gas: int = GAS_LIMIT, **credentials) -> str:
+    w3 = get_w3()
+
+    kvstore = w3.eth.contract(address=KVSTORE_CONTRACT, abi=kvstore_abi)
+    address_pk = kvstore.functions.get(GAS_PAYER, wallet_addr).call({
```
Go from `pk` to `pub_key`

gaieges replied:
sounds good

rbval commented on hmt_escrow/eth_bridge.py:
```diff
@@ -232,6 +238,46 @@ def deploy_factory(gas: int = GAS_LIMIT, **credentials) -> str:
+        'from':
+        GAS_PAYER
+    })
+    bytes_address = bytes(address_pk, encoding='utf-8')
```
`bytes_address` to `bytes_addr`

rbval commented on docker-compose.yml:
```diff
@@ -13,13 +13,15 @@ services:
     volumes:
-      - hmt_token_address:/deployed
-  
+      - hmt_token_address:/deployed-hmttoken
```
I'd like to avoid double t. `deployed_hmtoken` is prettier

gaieges replied:
ok

rbval commented on bin/wait_then_run:
```diff
@@ -5,14 +5,24 @@ JOB=${1-'python3 hmt_escrow/job.py -v'}
-until test -f /deployed/hmt.address.json; do
-  echo "waiting for hmt.address.json .."
+KVSTORE_ADDRESS_FILE=/deployed-kvstore/ethkvstore.address.json
+HMTTOKEN_ADDRESS_FILE=/deployed-hmttoken/hmt.address.json
```
avoid double t. `/deployed-hmtoken/hmt.address.json` is prettier

rbval commented on bin/wait_then_run:
```diff
@@ -5,14 +5,24 @@ JOB=${1-'python3 hmt_escrow/job.py -v'}
+KVSTORE_ADDRESS_FILE=/deployed-kvstore/ethkvstore.address.json
+HMTTOKEN_ADDRESS_FILE=/deployed-hmttoken/hmt.address.json
+
+until test -f $HMTTOKEN_ADDRESS_FILE; do
```
avoid double t and address. `$HMTOKEN_ADDR_FILE` is prettier.

gaieges commented:
Will fix up those renames .. lemme get this test working in the current form then I'll address the renames

rbval commented on hmt_escrow/eth_bridge.py:
```diff
@@ -232,6 +238,100 @@ def deploy_factory(gas: int = GAS_LIMIT, **credentials) -> str:
+    ValueError: environment variable GAS_PAYER required
+    >>> os.environ['GAS_PAYER'] = "0x1413862C2B7054CDbfdc181B83962CB0FC11fD92"
+    >>> os.environ['GAS_PAYER_PRIV'] = "28e516f1e2f99e96a48a23cea1f94ee5f073403a1c68e818263f0eb898f1c8e5"
+    >>> set_pub_key_at_addr('blah')  #doctest: +ELLIPSIS
```
Let's go with `b"2dbc2c2c86052702e7c219339514b2e8bd4687ba1236c478ad41b43330b08488c12c8c1797aa181f3a4596a1bd8a0c18344ea44d6655f61fa73e56e743f79e0d"` instead of `'blah'` as this is documentation at the same time.

gaieges replied:
done

gaieges commented:
> Last time I removed this I had problems in the consuming end

Removed what? the whole function? 

>  I'd probably test in bcs with this branches commit hash just to be sure.

Not a reason to hold this up, lets get it merged and we can pull it in

rbval commented on hmt_escrow/eth_bridge.py:
```diff
@@ -232,6 +238,100 @@ def deploy_factory(gas: int = GAS_LIMIT, **credentials) -> str:
+    >>> pub_key_to_set = b"2dbc2c2c86052702e7c219339514b2e8bd4687ba1236c478ad41b43330b08488c12c8c1797aa181f3a4596a1bd8a0c18344ea44d6655f61fa73e56e743f79e0d"
+    >>> set_pub_key_at_addr(pub_key_to_set)  #doctest: +ELLIPSIS
+    AttributeDict({'transactionHash': ...})
+    """
```
Nitpicking before perfect: one space needs to be added before closing docstring.

gaieges replied:
Done but we're not aiming for perfection here folks ..
